src/main/java/it/ristorantelorma/model/Restaurant.java

```diff
@@ -500,5 +501,113 @@ public static Result<Entry<Restaurant, Float>> getTopByNegativeReviews(
+         *
+         * @param connection la connessione al database
+         * @param restaurant il ristorante a cui aggiungere la vivanda
+         * @param name il nome della vivanda
+         * @param price il prezzo della vivanda
+         * @param type la tipologia della vivanda
+         * @return un Result contenente la vivanda aggiunta oppure un messaggio di errore
+         */
+        public static Result<Food> addFood(
+            final Connection connection,
+            final Restaurant restaurant,
+            final String name,
+            final BigDecimal price,
+            final FoodType type
+        ) {
+            try (
+                PreparedStatement statement = DBHelper.prepare(
+                    connection,
+                    "INSERT INTO vivande (nome, nome_attività, prezzo, tipologia) VALUES (?, ?, ?, ?);",
+                    name,
+                    restaurant.getRestaurantName(),
+                    price,
+                    type.getName()
+                );
+            ) {
+                    final int rows = statement.executeUpdate();
+                if (rows < 1) {
+                    return Result.failure("Vivanda non inserita.");
+                }
+                // Recupera la vivanda appena inserita
+                    final Result<Optional<Food>> foodRes = Food.DAO.find(connection, name, restaurant);
+                if (foodRes.isSuccess() && foodRes.getValue().isPresent()) {
+                    return Result.success(foodRes.getValue().get());
+                } else {
+                    return Result.failure("Vivanda inserita ma non trovata.");
+                }
+            } catch (SQLException e) {
+                final String errorMessage = "Errore nell'inserimento vivanda";
+                LOGGER.log(Level.SEVERE, errorMessage, e);
+                return Result.failure(errorMessage);
+            }
+        }
+
+        /**
+         * Elimina una vivanda tramite id.
+         *
+         * @param connection la connessione al database
+         * @param foodId l'id della vivanda da eliminare
+         * @return un Result che indica se l'eliminazione è avvenuta con successo oppure un messaggio di errore
+         */
+        public static Result<Boolean> deleteFood(
+            final Connection connection,
+            final int foodId
+        ) {
+            try (
+                PreparedStatement statement = DBHelper.prepare(
+                    connection,
+                    "DELETE FROM vivande WHERE codice = ?;",
+                    foodId
+                );
+            ) {
+                    final int rows = statement.executeUpdate();
+                return Result.success(rows > 0);
+            } catch (SQLException e) {
+                final String errorMessage = "Errore nell'eliminazione vivanda";
+                LOGGER.log(Level.SEVERE, errorMessage, e);
+                return Result.failure(errorMessage);
+            }
+        }
```

Comment:
why? `Food.DAO.list(Connection, Restaurant)` already exists, and the 2 other functions are misplaced (Food-related stuff goes in Food.java)